Build fails on OSX

```
$ npm --version
3.3.6
$ node --version
v4.1.1
$ node-gyp --version
v3.0.3
$ npm i robotjs
> robotjs@0.3.1 install /src/snip/node_modules/robotjs
> node-gyp rebuild

  CXX(target) Release/obj.target/robotjs/src/robotjs.o
../src/robotjs.cc:2:10: fatal error: 'nan.h' file not found
#include <nan.h>
         ^
1 error generated.
make: *** [Release/obj.target/robotjs/src/robotjs.o] Error 1
```

I even tried `npm i nan` on its own to no avail. Running on El Capitan.

Using `npm install git://github.com/octalmage/robotjs#dynamic-nan-include` gives me this error:

      CXX(target) Release/obj.target/robotjs/src/robotjs.o
    ../src/robotjs.cc:76:16: error: cannot initialize a variable of type 'MMMouseButton' (aka 'CGMouseButton') with an rvalue of type '_MMMouseButton'
            MMMouseButton button = LEFT_BUTTON;
                          ^        ~~~~~~~~~~~
    ../src/robotjs.cc:129:16: error: cannot initialize a variable of type 'MMMouseButton' (aka 'CGMouseButton') with an rvalue of type '_MMMouseButton'
            MMMouseButton button = LEFT_BUTTON;
                          ^        ~~~~~~~~~~~
    ../src/robotjs.cc:397:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
            *flags = MOD_ALT;
                   ^ ~~~~~~~
    ../src/robotjs.cc:401:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
            *flags = MOD_META;
                   ^ ~~~~~~~~
    ../src/robotjs.cc:405:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
            *flags = MOD_CONTROL;
                   ^ ~~~~~~~~~~~
    ../src/robotjs.cc:409:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
            *flags = MOD_SHIFT;
                   ^ ~~~~~~~~~
    ../src/robotjs.cc:413:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
            *flags = MOD_NONE;
                   ^ ~~~~~~~~
    ../src/robotjs.cc:425:13: error: cannot initialize a variable of type 'MMKeyFlags' (aka 'CGEventFlags') with an rvalue of type '_MMKeyFlags'
            MMKeyFlags flags = MOD_NONE;
                       ^       ~~~~~~~~
    ../src/robotjs.cc:479:13: error: cannot initialize a variable of type 'MMKeyFlags' (aka 'CGEventFlags') with an rvalue of type '_MMKeyFlags'
            MMKeyFlags flags = MOD_NONE;
                       ^       ~~~~~~~~

I saw this error before when I was running npm 2.x but I upgraded to 3.x and it changed to the `nan.h` error...

Same thing here.

```
$ npm i https://github.com/octalmage/robotjs.git\#dynamic-nan-include

> robotjs@0.3.1 install /src/snip/node_modules/robotjs
> node-gyp rebuild

  CXX(target) Release/obj.target/robotjs/src/robotjs.o
../src/robotjs.cc:76:16: error: cannot initialize a variable of type 'MMMouseButton' (aka 'CGMouseButton') with an rvalue of type '_MMMouseButton'
        MMMouseButton button = LEFT_BUTTON;
                      ^        ~~~~~~~~~~~
../src/robotjs.cc:129:16: error: cannot initialize a variable of type 'MMMouseButton' (aka 'CGMouseButton') with an rvalue of type '_MMMouseButton'
        MMMouseButton button = LEFT_BUTTON;
                      ^        ~~~~~~~~~~~
../src/robotjs.cc:397:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
        *flags = MOD_ALT;
               ^ ~~~~~~~
../src/robotjs.cc:401:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
        *flags = MOD_META;
               ^ ~~~~~~~~
../src/robotjs.cc:405:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
        *flags = MOD_CONTROL;
               ^ ~~~~~~~~~~~
../src/robotjs.cc:409:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
        *flags = MOD_SHIFT;
               ^ ~~~~~~~~~
../src/robotjs.cc:413:13: error: assigning to 'MMKeyFlags' (aka 'CGEventFlags') from incompatible type '_MMKeyFlags'
        *flags = MOD_NONE;
               ^ ~~~~~~~~
../src/robotjs.cc:425:13: error: cannot initialize a variable of type 'MMKeyFlags' (aka 'CGEventFlags') with an rvalue of type '_MMKeyFlags'
        MMKeyFlags flags = MOD_NONE;
                   ^       ~~~~~~~~
../src/robotjs.cc:479:13: error: cannot initialize a variable of type 'MMKeyFlags' (aka 'CGEventFlags') with an rvalue of type '_MMKeyFlags'
        MMKeyFlags flags = MOD_NONE;
                   ^       ~~~~~~~~
9 errors generated.
make: *** [Release/obj.target/robotjs/src/robotjs.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /src/snip/node_modules/robotjs
gyp ERR! node -v v4.1.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "https://github.com/octalmage/robotjs.git#dynamic-nan-include"
npm ERR! node v4.1.1
npm ERR! npm  v3.3.6
npm ERR! code ELIFECYCLE

npm ERR! robotjs@0.3.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the robotjs@0.3.1 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the robotjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls robotjs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /src/snip/npm-debug.log
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -2,7 +2,7 @@
   'targets': [{
     'target_name': 'robotjs',
     'include_dirs': [
-        'node_modules/nan/'
+        "<!(node -e \"require('nan')\")"
     ],
     
     'cflags': [

